### ISSUE TITLE
Problems with exceeded max_user_connections and timeout 300

### ISSUE THREAD

**Issue by banan44** [PrestaShop/PrestaShop#22882]
**Describe the bug**
Hello,
I have upgraded my shop from 1.7.6.9 to 1.7.7.0 a couple of days ago. Recently, my customers started to report problems with connection timeouts. I have CloudFlare service connected with my shop so the customers receive "Error 524 The origin web server timed out responding to this request". I have looked on logs and I noticed a few errors "Error on sending request ReceiveAckHdr: timeout 300 is exceeded"

I have tried to reproduce these problems. I went to Modules Manager and notices a few modules with a new version. Every time I clicked on upgrade button, the process took a few minutes and resulted with red popup on the corner with information about failed upgrading. When I refreshed the page I had Error 524. Here are the logs that appeared after the process. For security reasons I have changed some things in the location addresses:

```
[Mon Jan 18 18:16:09.641881 2021] [lsapi:error] [pid 22394:tid 140541547190016] [client 2a02:a314:8140:8380:154e:f5de:7a84:5ad:53500] [host domain.com] Error on sending request(POST /adminxxxx/index.php/improve/modules/manage/action/enable/psgdpr?_token=RYO0_ra-4kONS_V5ZwnW9itTbpx4CrJ9JSyjk-4kwLA HTTP/1.0); uri(/adminxxxx/index.php/improve/modules/manage/action/enable/psgdpr?_token=RYO0_ra-4kONS_V5ZwnW9itTbpx4CrJ9JSyjk-4kwLA) content-length(0): ReceiveAckHdr: timeout 300 is exceeded, referer: https://xxxx/adminxxxx/index.php/improve/modules/manage?_token=RYO0_ra-4kONS_V5ZwnW9itTbpx4CrJ9JSyjk-4kwLA

[Mon Jan 18 18:17:33.244893 2021] [lsapi:error] [pid 22680:tid 140541622724352] [client 2a02:a314:8140:8380:154e:f5de:7a84:5ad:37070] [host domain.com] Backend fatal error: PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000] [1226] User '224_presta' has exceeded the 'max_user_connections' resource (current value: 30) in /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:27\nStack trace:\n#0 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(27): PDO->__construct('mysql:host=loca...', '224_presta', 'Ff07b#AA1ulqc', Array)\n#1 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php(25): Doctrine\\DBAL\\Driver\\PDOConnection->__construct('mysql:host=loca...', '224_presta', 'Ff07b#AA1ulqc', Array)\n#2 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(356): Doctrine\\DBAL\\Driver\\PDOMySql\\Driver->connect(Array, '224_presta', 'Ff07b#AA1ulqc', Array)\n#3 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(422): Doctrine\\DBAL\\Connection->connect()\n#4 /home/websites/pres/vendor/doctri in /home/websites/pres/vendor/doctrine/doctrine-bundle/ConnectionFactory.php on line 79\n, referer: https://xxxx/adminxxxx/index.php/improve/modules/manage?_token=RYO0_ra-4kONS_V5ZwnW9itTbpx4CrJ9JSyjk-4kwLA

[Mon Jan 18 18:18:03.866847 2021] [lsapi:error] [pid 22397:tid 140541538797312] [client 2a02:a314:8140:8380:154e:f5de:7a84:5ad:37950] [host domain.com] Backend fatal error: PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000] [1226] User '224_presta' has exceeded the 'max_user_connections' resource (current value: 30) in /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:27\nStack trace:\n#0 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(27): PDO->__construct('mysql:host=loca...', '224_presta', 'Ff07b#AA1ulqc', Array)\n#1 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php(25): Doctrine\\DBAL\\Driver\\PDOConnection->__construct('mysql:host=loca...', '224_presta', 'Ff07b#AA1ulqc', Array)\n#2 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(356): Doctrine\\DBAL\\Driver\\PDOMySql\\Driver->connect(Array, '224_presta', 'Ff07b#AA1ulqc', Array)\n#3 /home/websites/pres/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(422): Doctrine\\DBAL\\Connection->connect()\n#4 /home/websites/pres/vendor/doctri in /home/websites/pres/vendor/doctrine/doctrine-bundle/ConnectionFactory.php on line 79\n

```

Everytime I had the error in logs the CPU usage of my server went from 20-30% to 150% (maximum allowed by the host). My host admin checked these logs and replied that the reason may be, for example, that some process on the website that takes a long time blocks the entire database, so that at that time other processes that are queued cannot be executed until the number of available connections to the database is exceeded. It would be necessary, therefore, to analyze the operation of the website and check what process is responsible for it. Is there a way to check all querries that are sent to database to find out which one takes too long and blocks the database?

**Steps to Reproduce**
In my situation go to Modules Manager and activate/deactivate module

**Screenshots**
No screenshot

**Additional information**
Server information Linux #1 SMP Thu Sep 17 06:10:33 EDT 2020 x86_64
Server software version: Apache
PHP version: 7.2.34
Memory limit: 800M
Max execution time: 600
Upload Max File size: 200M
MySQL version: 10.4.17-MariaDB-cll-lve
MySQL engine: InnoDB
MySQL driver: DbPDO

**Comment by hibatallahAouadni** [comment]
Hello @banan44 

We are aware of this issue, it is already fixed with our new patch realease 1.7.7.1. 
This issue is a duplicate of #22485. 
Please subscribe to the issue mentioned above.

PS: thanks for your detailed report :wink: 

Thanks!

**Comment by hibatallahAouadni** [comment]
Duplicate of #22485
